### ISSUE TITLE
Show Related Posts on Index and Archives

### ISSUE THREAD

**Issue by richardmtl** [Automattic/jetpack#409]
Related Posts only appear on individual posts. Can we add an option to make them appear on each post in index.php and archives? Similar to the option we have for sharing buttons. 

Originally requested here: http://wordpress.org/support/topic/related-posts-dont-show-up-on-main-page?replies=1

**Comment by mikeyarce** [comment]
Also suggested here: 2190849-t

**Comment by jeremyclarke** [comment]
Would like to add another dimension to this: Can we have the "related posts" display available outside of the "related posts" use-case? 

I love how they look and work. I would like to use the format in a widget on my homepage as the main headlines display for recent posts (the core Recent Posts widget is so crappy with no images etc). 

Would also be a great widget item to show off posts from a particular taxonomy term. 

Having a vertical format (for sidebars) would be pretty much mandatory to make  it work. 

I can create a new ticket for this, but figured I'd add it here to avoid adding noise.

**Comment by jerclarke** [comment]
I still want this feature, as well as my idea for a more generalized display!

**Comment by kraftbj** [comment]
Related Posts take the content of the current post, scores it, then compares that to other content on the site to come up with a sense of what's related. With having more "generic" related posts, what would y'all expect the "source" be to compare to?

It almost sounds like the type of display is notable and good, but the source of information may be different. Like, for an index or archive, maybe the most popular posts on a site/in a term may make mores sense?

This isn't something on the roadmap right now, but I appreciate your input so we have a better sense of it when it is up for consideration.

**Comment by jeherve** [comment]
Thanks for the extra details.

This sounds like a mix between Jetpack's [Top Posts Widget](https://jetpack.com/support/extra-sidebar-widgets/top-posts-pages-widget/), Jetpack's [Display Posts Widget](https://jetpack.com/support/extra-sidebar-widgets/display-wordpress-posts-widget/), and a yet to come Related Post widget. 

Widget may not be the correct term here, as it would be something you'd see as a shortcode as well. This may be a good candidate for a Gutenberg block maybe: basically a block offering you to display posts, with options like the number of posts you want to display, as well as how the posts should be queried: popularity, category / tag, and more.

Am I understanding things correctly here?

**Comment by kraftbj** [comment]
+1 to this being a potential future block.

**Comment by jerclarke** [comment]
@jeherve Yeah Gutenberg, I forgot about that. Having a block would be good too of course. I did mean widget as in widget though. I'd like to put recent posts, with that nice display in sidebars. 

You mention Top Posts and Display Posts widgets, which I don't use, but forgot why:

- `Top Posts & Pages` has "text list" "image list" and "image grid", but all three are very flawed compared to what I see from "related posts". 
   - The text list is the same as the default WP recent posts widget
   - the image list uses tiny images
   - the image grid is JUST images, when what would make a nice consistent UI is a grid of image+headline / image+headline / image+headline

`Display WordPress Posts` I honestly don't know what it does because I can't get it to work but it seems to demand a URL of an external WP site, which isn't what I want. When I give it my own website's address (which is silly but I tried it) I get an error: 

> An error has occurred. Please reload the page and try again.

So maybe `Display WordPress Posts` does look and work like "Related Posts", but if it does, I think it should be rebranded as "Display Posts" and the "external" aspect should be optional (and it should work better and fail with more specific errors). "Display WordPress Posts" is pretty ambiguous since it could also describe local posts. 

Ultimately I think it would be ideal if there were less Jetpack widgets, because there are so so many now, and it's confusing. If there was one called "Display Posts (Jetpack)", it could have options for "top posts", post types, taxonomies etc. all in one place and with a consistent set of display modes (text/small image/large image/just image).

**Comment by jeherve** [comment]
> I honestly don't know what it does because I can't get it to work

It should work with any URL of a WordPress.com or Jetpack site, even your own. Small caveat; if using a Jetpack site, the JSON API module has to be activated. It is activated by default though, so that should not be an issue.

If you'd like we can take a closer look as to why that does not work for you; don't hesitate to [send us an email](https://jetpack.com/contact-support/?rel=support) with more details about the site, or you can just bug Richard A. about it the next time you meet him :)

> If there was one called "Display Posts (Jetpack)", it could have options for "top posts", post types, taxonomies etc. all in one place and with a consistent set of display modes (text/small image/large image/just image).

That makes sense. It would certainly be ideal and much simpler! It may not be easy though, as many people use the existing widgets and sometimes heavily customize them. As you pointed out the look may not be what you're looking for, which is why some people customize them through filters (to get bigger images for example) or CSS.

**Comment by tyxla** [comment]
We're planning to add Related Posts as a block in the next Jetpack version - 6.9. https://github.com/Automattic/jetpack/pull/10132 has already been merged. Should we close this issue, @jeherve?

**Comment by jeherve** [comment]
Yes, I'll remove that label. The block will address some of this by offering a way for folks to show Related Posts on archive pages if they add the block to a post that is displayed there.
